### DISCUSSION TITLE
How to prevent re-renders when working with array of items?

### DISCUSSION THREAD

**Discussion by bartekczyz** [pmndrs/zustand#798] [Q&A] [answered]
Hello, I have a problem with the component I'm implementing - `CheckboxGroup`. When I check a checkbox, all other items are re-rendered even though its values shouldn't change. Try to select an item and look at the console. I've created a wrapper component with `React.memo`, but it didn't prevent it from re-rendering. I also have a similar component - `RadioGroup`, but it works like a charm. I suspect that's because it holds only one primitive value while `CheckboxGroup` holds an array. 

Any help will be appreciated. Thanks!

Here's a repro: https://codesandbox.io/s/crazy-northcutt-903cc

**Comment by dai-shi** [comment, marked as answer]
Yeah, `active` is an array changing for all items.
Quick fix: https://codesandbox.io/s/hopeful-leavitt-kg06x?file=/src/store.ts

**Reply by bartekczyz** [reply]
Awesome, thank you!

**Reply by eitai** [reply]
hi i got the same issue can u give me a lil explation whats going on ?
